backend/internal/userschema/service/userschemaservice.go

```diff
@@ -0,0 +1,505 @@
+		if !allowedFields[field] {
+			return fmt.Errorf("invalid field '%s' for object property", field)
+		}
+	}
+
+	properties, exists := propMap["properties"]
+	if !exists {
+		return fmt.Errorf("missing required 'properties' field for object type")
+	}
+
+	propertiesMap, ok := properties.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("'properties' field must be an object")
+	}
+
+	for nestedPropName, nestedPropDef := range propertiesMap {
+		if err := validateUserSchemaProperty(nestedPropDef); err != nil {
+			return fmt.Errorf("invalid nested property '%s': %w", nestedPropName, err)
+		}
+	}
+
+	return nil
+}
+
+// validateArrayProperty validates array property definitions.
+func validateArrayProperty(propMap map[string]interface{}) error {
+	items, exists := propMap["items"]
+	if !exists {
+		return fmt.Errorf("missing required 'items' field for array type")
+	}
+
+	allowedFields := map[string]bool{
+		"type":  true,
+		"items": true,
+	}
+	for field := range propMap {
+		if !allowedFields[field] {
+			return fmt.Errorf("invalid field '%s' for array property", field)
+		}
+	}
+
+	itemsMap, ok := items.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("'items' field must be an object")
+	}
+
+	itemType, exists := itemsMap["type"]
+	if !exists {
+		return fmt.Errorf("missing required 'type' field in items definition")
+	}
+
+	itemTypeStr, ok := itemType.(string)
+	if !ok {
+		return fmt.Errorf("'type' field in items must be a string")
+	}
+
+	switch itemTypeStr {
+	case schema.TypeString, schema.TypeNumber, schema.TypeBoolean:
+		return validateArrayLeafItems(itemTypeStr, itemsMap)
+	case schema.TypeObject:
+		return validateObjectProperty(itemsMap)
+	default:
+		return fmt.Errorf("invalid items type '%s', must be one of: string, number, boolean, object", itemTypeStr)
+	}
+}
+
+// validateArrayLeafItems validates array items for leaf types (string, number, boolean).
+func validateArrayLeafItems(itemType string, itemsMap map[string]interface{}) error {
+	itemEnum, exists := itemsMap["enum"]
+	if exists {
+		itemEnumArray, ok := itemEnum.([]interface{})
+		if !ok {
+			return fmt.Errorf("'enum' field in items must be an array")
+		}
+		if len(itemEnumArray) == 0 {
+			return fmt.Errorf("'enum' array in items cannot be empty")
+		}
+		return validateEnumItemsType(itemType, itemEnumArray)
+	}
+	return nil
+}
+
+// validateEnumItemsType validates array enum items for leaf types (string, number, boolean).
+func validateEnumItemsType(expectedType string, enumArray []interface{}) error {
+	for i, item := range enumArray {
+		switch expectedType {
+		case schema.TypeString:
+			if _, ok := item.(string); !ok {
+				return fmt.Errorf("'enum' array item at index %d must be a string to match property type", i)
+			}
+		case schema.TypeNumber:
+			switch item.(type) {
+			case float64, int:
+				// Valid number type
+			default:
+				return fmt.Errorf("'enum' array item at index %d must be a number to match property type", i)
+			}
```

Comment:
Suggestion: wouldn't it better to have a simple if condition here?

Reply:
then, it will be look like this:
```
			if _, ok := item.(float64); !ok {
				if _, ok := item.(int); !ok {
					return fmt.Errorf("'enum' array item at index %d must be a number to match property type", i)
				}
			}
```

or

```
			if _, ok := item.(float64); !ok && func() bool { _, ok := item.(int); return !ok }() {
				return fmt.Errorf("'enum' array item at index %d must be a number to match property type", i)
			}
```

the current way is simpleter, compared to the above ones.